src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -234,41 +252,63 @@ impl LeaderBlockCommitOp {
             return Err(op_error::ParseError);
         }
 
-        let mut commit_outs = vec![];
-        let mut pox_fee = None;
-        for (ix, output) in outputs.into_iter().enumerate() {
-            // only look at the first OUTPUTS_PER_COMMIT outputs
-            if ix >= OUTPUTS_PER_COMMIT {
-                break;
+        // check if we've reached PoX disable
+        let (commit_outs, sunset_burn, burn_fee) = if block_height >= pox_sunset_ht {
+            if !outputs[0].address.is_burn() {
+                return Err(op_error::BlockCommitBadOutputs);
             }
-            // all pox outputs must have the same fee
-            if let Some(pox_fee) = pox_fee {
-                if output.amount != pox_fee {
-                    warn!("Invalid commit tx: different output amounts for different PoX reward addresses");
-                    return Err(op_error::ParseError);
+            let BurnchainRecipient { address, amount } = outputs.remove(0);
+            (vec![address], 0, amount)
+        } else {
+            // check if this transaction provided a sunset burn
+            let sunset_burn = if let Some(sunset_burn_output) = outputs.get(OUTPUTS_PER_COMMIT) {
```

Comment:
I take it we couldn't burn BTC by sending it to the `OP_RETURN`?

Reply:
Oh, shoot, I completely forgot about that 😞  Can update.

Reply:
Updated in b9af529